rules/aws_s3_rules/aws_s3_bucket_defacement_attempt.yml

```diff
@@ -0,0 +1,82 @@
+AnalysisType: rule
+Description: Adversaries may modify visual content available internally or externally to an enterprise network, thus affecting the integrity of the original content. Reasons for Defacement include delivering messaging, intimidation, or claiming (possibly false) credit for an intrusion. Disturbing or offensive images may be used as a part of Defacement in order to cause user discomfort, or to pressure compliance with accompanying messages.
+DisplayName: S3 Bucket Defacement Attempt
+Enabled: false
+Filename: aws_s3_bucket_defacement_attempt.py
+Reference: https://attack.mitre.org/techniques/T1491/
+Reports:
+    MITRE ATT&CK:
+        - TA0040:T1491
+Severity: Low
+Tests:
+    - ExpectedResult: true
```

Comment:
This detection should include test cases that show

1.  S3 Get operation which does not match the EXCLUDE_* rules -> false
2.  At least one test case demonstrating that the EXCLUDE_* functions ->  false 
3. a s3 put operation which returns  -> true. You'll need to [mock](https://github.com/panther-labs/panther-analysis/blob/master/rules/aws_cloudtrail_rules/aws_console_login_without_mfa.yml#L405) the INCLUDED_BUCKET_NAMES object. I'm happy to help with that if you would like 👍

Reply:
Hey hey! I made some changes and pushed them. However, it doesn't appear to pass linting:

> rules/aws_s3_rules/aws_s3_bucket_defacement_attempt.py:22:0: R1260: 'rule' is too complex. The McCabe rating is 15 (too-complex)
> rules/aws_s3_rules/aws_s3_bucket_defacement_attempt.py:22:0: R0911: Too many return statements (7/6) (too-many-return-statements)
> rules/aws_s3_rules/aws_s3_bucket_defacement_attempt.py:22:0: R0912: Too many branches (14/12) (too-many-branches)

Are those thresholds arbitrary?

Reply:
Those thresholds are intentional ( arbitrary is a word that might be universally applied to any style/lint guidelines ). There's ~1 detection that's disabling these by intention. I can take a look in here and see if there's a way to logically simplify the detection